src/Asm89/Stack/CorsService.php

```diff
@@ -174,9 +174,16 @@ private function checkOrigin(Request $request)
             // allow all '*' flag
             return true;
         }
+
         $origin = $request->headers->get('Origin');
 
-        return in_array($origin, $this->options['allowedOrigins']);
```

Comment:
I'd rather keep this the default, but I can see a case for people wanting to use the more involved `OriginMatcher`. How about adding a new optional constructor argument `$originChecker` and a new interface `interface OriginChecker { function check(string $origin): bool; }`. If `$this->originChecker` is `null` we'll use the current codepath. If it's set to something custom we'll use that instead.